### PR TITLE
Added new invite fields.

### PR DESCRIPTION
**Pull request by vicky5124**
Adds new fields documented in [here](https://discord.com/developers/docs/resources/invite#invite-object) and [here](https://discord.com/developers/docs/resources/channel#create-channel-invite)

	modified:   src/builder/create_invite.rs
	modified:   src/model/invite.rs

### CODE REVIEW

File: src/model/invite.rs [hunk 1]
```diff
@@ -404,3 +422,33 @@ impl RichInvite {
...
+    participant_count: u64,
+    /// The number of users speaking in the Stage
+    speaker_count: u64,
+    /// The topic of the Stage instance (1-120 characters)
+    topic: String,
+}
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+#[non_exhaustive]
+#[cfg(feature = "model")]
```
Comment (arqunis) [inline src/model/invite.rs]: The `model` feature only applies to methods on model structs. Feature-gating of the `model` feature shouldn't occur on structs nor enums.
```suggestion
```

File: src/model/invite.rs [hunk 2]
```diff
@@ -404,3 +422,33 @@ impl RichInvite {
         format!("https://discord.gg/{}", self.code)
     }
 }
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg(feature = "model")]
```
Comment (arqunis) [inline src/model/invite.rs]:
```suggestion
```

File: src/builder/create_invite.rs [hunk 3]
```diff
@@ -206,6 +208,41 @@ impl CreateInvite {
         self.0.insert("unique", Value::Bool(unique));
         self
     }
+
+    /// The type of target for this voice channel invite.
+    #[cfg(feature = "model")]
```
Comment (arqunis) [inline src/builder/create_invite.rs]:
```suggestion
```

**Review by arqunis — CHANGES_REQUESTED** [review]

File: src/model/invite.rs [hunk 4]
```diff
@@ -46,6 +46,24 @@ pub struct Invite {
...
+    /// Only shows up if `target_type` is `Stream`.
+    pub target_user: Option<UserId>,
+    /// The embedded application to open for this voice channel embedded application invite.
+    ///
+    /// Only shows up if `target_type` is `EmmbeddedApplication`.
+    pub target_application: Option<ApplicationId>,
+
+    // /// The expiration date of this invite, returned from `Http::get_invite` when
+    // /// `with_expiration` is true.
+    // pub expires_at: Option<DateTime<Utc>>,
```
Comment (arqunis) [inline src/model/invite.rs:63]: I presume the extra slashes are accidental?
```suggestion
    /// The expiration date of this invite, returned from `Http::get_invite` when
    /// `with_expiration` is true.
    pub expires_at: Option<DateTime<Utc>>,
```

**Review by arqunis — CHANGES_REQUESTED** [review]

File: src/model/invite.rs (hunk 4 above)
Comment (vicky5124) [reply src/model/invite.rs:63]: No, the Http method doesn't support providing that field, so I commented it out.
I wanted this to be non-breaking, and if expires_at would always be None, there's no point in implementing it. The code is still there for some possible breaking change though (different PR after this?)

**Review by arqunis — APPROVED** [review]

**Pull request merged**